src/main/scala/coreplex/Coreplex.scala

```diff
@@ -49,7 +50,9 @@ case class CoreplexConfig(
   val plicKey = PLICConfig(nTiles, hasSupervisor, nExtInterrupts, 0)
 }
 
-abstract class Coreplex(implicit val p: Parameters, implicit val c: CoreplexConfig) extends Module
+abstract class Coreplex(clockSignal: Clock = null, resetSignal: Bool = null)
+    (implicit val p: Parameters, implicit val c: CoreplexConfig)
+    extends Module(Option(clockSignal), Option(resetSignal))
```

Comment:
I'm torn about both line breaks here.  Maybe using _clock and _reset instead would let you put the (implicit val ...) on the same line as Coreplex?